src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs

```diff
@@ -417,6 +418,110 @@ public async Task CreateOrUpdatePullRequestCommentAsync(string pullRequestUrl, s
             }
         }
 
+        public async Task CreateOrUpdatePullRequestMergeStatusInfoAsync(string pullRequestUrl, IReadOnlyList<MergePolicyEvaluationResult.SingleResult> evaluations)
+        {
+            (string owner, string repo, int id) = ParsePullRequestUri(pullRequestUrl);
+
+            // Get the latest commit for the current PR
+            IList<Commit> commits = await GetPullRequestCommitsAsync(pullRequestUrl);
+            Commit latestCommit = commits[commits.Count - 1];
+
+            // Get all the checks runs for the current PR
+            CheckRunsResponse existingChecksRuns = await Client.Check.Run.GetAllForReference(owner,repo, latestCommit.Sha);
+            // Convert the IReadOnlyList of CheckRun to a List of CheckRun
+            List<CheckRun> existingChecksList = new List<CheckRun>(existingChecksRuns.CheckRuns);
+
+            foreach (var eval in evaluations)
+            {
+                CheckRun existingCheckRun = existingChecksRuns.CheckRuns.SingleOrDefault(c => c.ExternalId == $"maestro-policy-{eval.MergePolicyName}-{latestCommit.Sha}");
+                NewCheckRun newCheckRun = CreateNewCheckRun(eval, latestCommit.Sha);
+                // If the check doesn't exist yet, create it
+                if (existingCheckRun == null)
+                {
+                    await Client.Check.Run.Create(owner,repo, newCheckRun);
+                }
+
+                // If the check exist, checks that the status are different to update it
+                else if (existingCheckRun != null && newCheckRun.Status != existingCheckRun.Status)
+                {
+                    CheckRunUpdate updatedCheck = new CheckRunUpdate();
+                    updatedCheck.Status = newCheckRun.Status;
+                    updatedCheck.Conclusion = newCheckRun.Conclusion;
+                    updatedCheck.Name = newCheckRun.Name;
+                    updatedCheck.CompletedAt = newCheckRun.CompletedAt;
+                    existingChecksList.Remove(existingCheckRun);
+                    await Client.Check.Run.Update(owner,repo, existingCheckRun.Id, updatedCheck);
+                }
+            }
+
+            // Remove the check(s) in existingChecksRuns that aren't in evaluations 
+            foreach (var remainingCheck in existingChecksList)
+            {
+                MergePolicyEvaluationResult.SingleResult remainingCheckCommon = evaluations.SingleOrDefault(eval => remainingCheck.ExternalId == $"maestro-policy-{eval.MergePolicyName}-{latestCommit.Sha}");
+                // Avoid deleting check(s) that aren't from maestro
+                if (remainingCheckCommon != null)
```

Comment:
remainingCheck should never be in evaluations, so this will always be false. Consider to use  `if (remainingCheck.ExternalId.StartsWith("maestro-policy-"))`